### DISCUSSION TITLE
"thread failed to start" on Windows when using clumsy

### DISCUSSION THREAD

**Discussion by fengdyli** [curl/curl#11349] [Q&A] [answered]
In the case of poor networking, curl_easy_perform and curl_multi_perform will be stuck resolving dns, and this time cannot be fixed. I want to set the timeout period for resolving dns.

**Comment by bagder** [comment, marked as answer]
I recommend using one of both of these: [CURLOPT_TIMEOUT_MS](https://curl.se/libcurl/c/CURLOPT_TIMEOUT_MS.html) and [CURLOPT_CONNECTTIMEOUT_MS](https://curl.se/libcurl/c/CURLOPT_CONNECTTIMEOUT_MS.html)

**Reply by fengdyli** [reply]
That doesn't work, you can test with clusmy, and change all the numbers in clusmy to highest.
for example, I set CURLOPT_TIMEOUT_MS 5000ms, and set CURLOPT_CONNECTTIMEOUT_MS 1000ms, start clusmy, run the code. Got an error: getaddrinfo() thread failed to start. not Connection Timeout after ...ms.
if you get error: Connection timeout after ...ms, you can run `ipconfig /flushdns` in cmd, and run your code again.
the error `getaddrinfo() thread fialed to start` appeared more than 5000ms

**Reply by bagder** [reply]
"clumsy" seems to be a Windows application so I cannot "test with it" easily as I don't use Windows.

> Got an error: getaddrinfo() thread failed to start

That is unexpected and should not happen. What curl version are you using? Are you running an anti-virus or something that might inspect/break local socket traffic?

If it can't start the name resolver thread, then things don't work "normally" and then name resolve timeouts will not work.

**Reply by fengdyli** [reply]
I am using 7.80 version.
Clumsy is a software that makes the net poor, I use it to test some extreme network scenarios. If i close it, the URL is accessible, so it's not a problem with my computer settings, and I think curl has no question, but it lacks a timeout interface for resovle DNS

**Reply by bagder** [reply]
It seems you're not reading what I write. It does not lack timeout options. It has several, as I already mentioned.

Your problem is that curl fails with `getaddrinfo() thread failed to start` and therefore it has crippled functionality - the timeout function cannot timeout the resolve when this happens. **That** is the issue to figure out.

**Reply by fengdyli** [reply]
I know what you say, `getaddrinfo() thread failed to start` was happened when I open clumsy, this error is related to DNS resolution. So in other words,, when i use curl in poor net, it will stuck on resovling DNS, the option [CURLOPT_TIMEOUT_MS](https://curl.se/libcurl/c/CURLOPT_TIMEOUT_MS.html) and [CURLOPT_CONNECTTIMEOUT_MS](https://curl.se/libcurl/c/CURLOPT_CONNECTTIMEOUT_MS.html) doesn't work.

**Reply by bagder** [reply]
Right, so for some reason clumsy breaks curl so that it can't run its name resolver correctly. This might be a curl bug, it could also be a clumsy bug.

**Reply by fengdyli** [reply]
As i said, the clumsy just make a extreme network scenarios.In a real situation, some users with bad networks get stuck parsing DNS.

**Reply by bagder** [reply]
We don't see people report this problem though so this is a strong reason to believe this is *made to happen* by clumsy. So a real world problem will get a proper timeout. But not with clumsy.

Someone on Windows using clumsy needs to debug this.

**Reply by fengdyli** [reply]
So, I am the only guy
